pkg/expect/expect.go

```diff
@@ -193,7 +193,7 @@ func (ep *ExpectProcess) ExpectFunc(ctx context.Context, f func(string) bool) (s
 
 		select {
 		case <-ctx.Done():
-			return "", fmt.Errorf("context done before matching log found")
+			return "", fmt.Errorf("no matching log found: %w", ctx.Err())
```

Comment:
I'm not entirely sure if this change is ok. Previsouly the `context.DeadlineExceeded` error was "swallowed" here. 

This was not discovered until now because the test for this wrongly used `require.ErrorAs` (not appropriate for a sentinel error value like `context.DeadlineExceeded`) instead of `require.ErrorIs`

Reply:
I do not see any issue to wrap the err as long as all test can pass. But I suggest not to change the error message.

Reply:
👍🏻 I updated the PR and left the message unchanged but wrapped the error. Tests seem to pass.